x/spec/types/spec.go

```diff
@@ -7,6 +7,97 @@ import (
 
 const minCU = 1
 
+type getterFunc func(index string) (Spec, bool)
+
+// ExpandSpec expands the "imports" field of the spec -if needed, recursively-
+// and adds to the current Spec those additional APIs from the imported Spec(s).
+// It takes as argument a getter function to fetch imported Spec(s) by name from
+// the keeper store, on demand. (Can't import the keeper module due to circular
+// dependencies)
+func (spec *Spec) ExpandSpec(getSpec getterFunc) (map[string]string, error) {
+	depends := map[string]bool{spec.Index: true}
```

Comment:
refactor: this is a confusing layout and complex implementation, passing callbacks without added value (it's always the same one too). instead of adding it as a method to the spec object and then have circular dependencies, if it accesses the keeper it should be implemented as a keeper function

Reply:
Done.